docs/components/compact.md

```diff
@@ -279,6 +279,9 @@ usage: thanos compact [<flags>]
 Continuously compacts blocks in an object store bucket.
 
 Flags:
+      --auto-gomemlimit.ratio=0.9
```

Comment:
Yes definitely!
Just raise this in compactor and if the initial implementation is fine, I'll add to the other components

Reply:
> Maybe we could enable this functionality automatically if ratio > 0? 

The same GOMEMLIMIT feature in Prometheus works with the ratio being defaulted to 0.9 (as is here), and applied/ignored based on the enable flag state. To me it would make sense to follow the same logic, instead of inferring the enable flag state from the ratio value. (--> leave it as it is now)

https://github.com/prometheus/prometheus/blob/69ce3780d7f367e11e671d725fafbda00e2a2f87/cmd/prometheus/main.go#L728

my 2 cents

Reply:
indeed would be better some consistency, is that ok for you @GiedriusS ?

Reply:
Ahh, it makes sense.

Reply:
Cool!
I'll add to the other commands and push it.